spec/validators/evidence_held_validator_spec.rb

```diff
@@ -0,0 +1,290 @@
+RSpec.describe EvidenceHeldValidator, type: :model do
+  subject { klass.new(declaration_type:, evidence_held:, training_period:) }
+
+  let(:klass) do
+    Class.new do
+      include ActiveModel::Model
+      include ActiveModel::Validations
+
+      validates :evidence_held, evidence_held: true
+
+      attr_accessor :declaration_type, :evidence_held, :training_period
+
+      def self.model_name
+        ActiveModel::Name.new(self, nil, "temp")
+      end
+
+      def initialize(declaration_type:, evidence_held:, training_period:)
+        @declaration_type = declaration_type
+        @evidence_held = evidence_held
+        @training_period = training_period
+      end
+    end
+  end
+  let(:contract_period) { Struct.new(:detailed_evidence_types_enabled).new(detailed_evidence_types_enabled) }
+  let(:training_period) { Struct.new(:for_ect?, :contract_period).new(trainee_type == :ect, contract_period) }
+
+  describe "#validate" do
+    context "ECT participant" do
+      let(:trainee_type) { :ect }
+
+      context "when contract period has simple evidence types" do
+        let(:detailed_evidence_types_enabled) { false }
+
+        context "when `declaration_type` is `started`" do
+          let(:declaration_type) { "started" }
+
+          context "when `evidence_held` is nil" do
+            let(:evidence_held) { nil }
+
+            it "does not show validation error" do
+              expect(subject).to be_valid
+            end
+          end
+
+          context "when `evidence_held` is invalid" do
+            let(:evidence_held) { "one-term-induction" }
+
+            it "does not show validation error" do
+              expect(subject).to be_valid
+            end
+          end
+
+          context "when `evidence_held` is `other`" do
+            let(:evidence_held) { "other" }
+
+            it "does not show validation error" do
+              expect(subject).to be_valid
+            end
+          end
+        end
+
+        context "when `declaration_type` is other than started" do
+          let(:declaration_type) { "retained-1" }
+
+          context "when `evidence_held` is nil" do
+            let(:evidence_held) { nil }
+
+            it "has a meaningful error", :aggregate_failures do
+              expect(subject).to be_invalid
+              expect(subject.errors.messages_for(:evidence_held)).to include("Enter a '#/evidence_held' value for this participant.")
```

Comment:
Is it worth using `contain_exactly` so we verify this is the only error message? 💭